### PR TITLE
Rename indexExchange to ix (#657)

### PR DESCRIPTION
**Pull request by ix-certification**
- Fixes the issues addressed in #657.
- Rename package from `indexExchange` to `ix`
- Fixes `siteId` being incorrect type, Prebid.js sends as a string.
- We do not currently support video for Prebid Server.
- Added default endpoint for ix.
- Response will match ad unit code based on bid id.

### CODE REVIEW

**Comment by dbemiller** [comment]
@ix-certification This will break far too many things :/.

Prebid Mobile and AMP use Stored Requests. Every PBS host company has their own data, and all that data will still use `indexExchange`. If you just delete it, suddenly `indexExchange` will be an unrecognized Bidder, and everyone's pages will break.

We could deprecate it, and delete it in the distant future (after everyone host PBS has migrated their DB data)... but I don't think it's fair to delete it suddenly and with no window for migration.

**Comment by dbemiller** [comment]
One thing @mkendall07 did discover this morning: according to our metrics, `indexExchange` returns _no_ valid bids at all, to anyone.

So another option is that we could relax the input validation. Today, PBS returns a 4xx if you use an "unknown" bidder. It was designed that way to catch dumb typos or missing aliases.... but if we're ok losing that, we could run the other bidders and return a `req.ext` error describing the wrong bidder instead. That change would be backwards compatible, and once we made them, these changes wouldn't really "break" anything, since `indexExchange` isn't bidding today anyway.

**Comment by hhhjort** [comment]
After some offline discussion, it seems dbemiller's last idea is the best. With a slight modification, we will relax the constraint to not 4xx any "deprecated" bidders, which will include the old name for the index exchange bidder, while leaving the general case intact (truely unknown bidders will still cause a 4xx). Once that PR goes in, we can merge this if no other issues come up in review.

**Comment by hhhjort** [comment]
Submitted PR #716 in support of this PR, so existing requests won't break when the name change happens.

File: adapters/ix/ix.go [hunk 1]
```diff
@@ -17,31 +17,31 @@ import (
...
+type IxAdapter struct {
 	http *adapters.HTTPAdapter
 	URI  string
 }
 
 // used for cookies and such
-func (a *IndexAdapter) Name() string {
-	return "indexExchange"
+func (a *IxAdapter) Name() string {
+	return "ix"
```
Comment (dbemiller) [inline adapters/ix/ix.go]: Should return the constant, `openrtb_ext.BidderIx` here.

File: adapters/ix/ix.go [hunk 2]
```diff
@@ -136,38 +136,40 @@ func (a *IndexAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder *pb
...
+						BidID:             bidID,
+						AdUnitCode:        adunit.Code,
+						BidderCode:        bidder.BidderCode,
+						Price:             bid.Price,
+						Adm:               bid.AdM,
+						Creative_id:       bid.CrID,
+						Width:             bid.W,
+						Height:            bid.H,
+						DealId:            bid.DealID,
+						CreativeMediaType: "banner",
```
Comment (dbemiller) [inline adapters/ix/ix.go]: Should use the `openrtb_ext.BidTypeBanner` constant here.

**Pull request merged**